api/src/wfl/module/covid.clj

```diff
@@ -94,82 +96,122 @@
+             (if (and started (not finished)) (update! workload (utc-now)) workload)))
 
 (defn ^:private stop-covid-workload
-  "Use transaction `tx` to stop the `workload` looking for new data."
-  [tx {:keys [started stopped finished] :as workload}]
-  (letfn [(stop! [{:keys [id source] :as workload} now]
-            (source/stop-source! tx source)
-            (patch-workload tx workload {:stopped now :updated now})
-            (when-not (:started workload)
-              (patch-workload tx workload {:finished now}))
-            (workloads/load-workload-for-id tx id))]
-    (when-not started
-      (throw (UserException. "Cannot stop workload before it's been started."
-                             {:workload workload})))
-    (if-not (or stopped finished) (stop! workload (utc-now)) workload)))
+      "Use transaction `tx` to stop the `workload` looking for new data."
+      [tx {:keys [started stopped finished] :as workload}]
+      (letfn [(stop! [{:keys [id source] :as workload} now]
+                     (source/stop-source! tx source)
+                     (patch-workload tx workload {:stopped now :updated now})
+                     (when-not (:started workload)
+                               (patch-workload tx workload {:finished now}))
+                     (workloads/load-workload-for-id tx id))]
+             (when-not started
+                       (throw (UserException. "Cannot stop workload before it's been started."
+                                              {:workload workload})))
+             (if-not (or stopped finished) (stop! workload (utc-now)) workload)))
 
 (defn ^:private workload-to-edn [workload]
-  (-> workload
-      (util/select-non-nil-keys workload-metadata-keys)
-      (dissoc :pipeline)
-      (update :source   util/to-edn)
-      (update :executor util/to-edn)
-      (update :sink     util/to-edn)))
-
-(defoverload workloads/create-workload!    pipeline create-covid-workload)
-(defoverload workloads/start-workload!     pipeline start-covid-workload)
-(defoverload workloads/update-workload!    pipeline update-covid-workload)
-(defoverload workloads/stop-workload!      pipeline stop-covid-workload)
-(defoverload workloads/retry               pipeline batch/retry-unsupported)
-(defoverload workloads/load-workload-impl  pipeline load-covid-workload-impl)
-(defmethod   workloads/workflows           pipeline
-  [tx {:keys [executor] :as _workload}]
-  (executor/executor-workflows tx executor))
-(defmethod   workloads/workflows-by-status pipeline
-  [tx {:keys [executor] :as _workload} status]
-  (executor/executor-workflows-by-status tx executor status))
-(defoverload workloads/to-edn             pipeline workload-to-edn)
+      (-> workload
+          (util/select-non-nil-keys workload-metadata-keys)
+          (dissoc :pipeline)
+          (update :source util/to-edn)
+          (update :executor util/to-edn)
+          (update :sink util/to-edn)))
+
+(defn retry-workflows
```

Comment:
(A general note so that I don't forget, not specific to your debugging today)

Ultimately the code that you're writing here looks like it should live in the `executor` namespace.  None of it seems COVID-specific, or code that bridges different stages.  A tip-off is how you've publicized several `executor` methods to make this possible.

Reply:
Agreed. It was always my intention to make this generalized. I just have been doing my work in the covid file, but should probably move it to the executor file.

Reply:
Actually, if you want to hold off on that move, I should have a PR for executor logging tomorrow!  Then we won't step on each other's toes.